warden/group/handler.go

```diff
@@ -172,19 +168,14 @@ func (h *Handler) FindGroupNames(w http.ResponseWriter, r *http.Request, member
 //       403: genericError
 //       500: genericError
 func (h *Handler) CreateGroup(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
-	var g Group
-	var ctx = r.Context()
-
-	if err := json.NewDecoder(r.Body).Decode(&g); err != nil {
-		h.H.WriteError(w, r, errors.WithStack(err))
+	if err := h.checkRequest(r, h.PrefixResource(GroupsResource), "create"); err != nil {
```

Comment:
isn't it better to always first check the request?

Reply:
Yes, maybe I was using `g` here for some reason which was why it was decoded first, but as it stands now the order can be reversed´the way you did it.